xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -741,6 +750,93 @@ def test_publish_automatically_after_delete_unit(self, default_ms):
         self.store.delete_item(vertical.location, self.user_id)
         assert not self._has_changes(sequential.location)
 
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_course_create_event(self, default_ms):
+        """
+        Check that COURSE_CREATED event is sent when a course is created.
+        """
+        self.initdb(default_ms)
+
+        event_receiver = Mock()
+        COURSE_CREATED.connect(event_receiver)
+
+        test_course = self.store.create_course('test_org', 'test_course', 'test_run', self.user_id)
+
+        event_receiver.assert_called()
+
+        self.assertDictContainsSubset(
+            {
+                "signal": COURSE_CREATED,
+                "sender": None,
+                "course": CourseData(
+                    course_key=test_course.id,
+                ),
+            },
+            event_receiver.call_args.kwargs
+        )
+
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_xblock_create_event(self, default_ms):
+        """
+        Check that XBLOCK_CREATED event is sent when xblock is created.
+        """
+        self.initdb(default_ms)
+
+        event_receiver = Mock()
+        XBLOCK_CREATED.connect(event_receiver)
+
+        xblock = self.store.create_item(
+            self.user_id,
+            self.course.id,
+            'vertical',
+            block_id='parent',
+        )
+
+        event_receiver.assert_called()
+        self.assertDictContainsSubset(
+            {
+                "signal": XBLOCK_CREATED,
+                "sender": None,
+                "xblock_info": XBlockData(
+                    usage_key=xblock.location,
+                    block_type=xblock.location.block_type,
+                ),
+            },
+            event_receiver.call_args.kwargs
+        )
```

Comment:
This is not passing.

```
 AssertionError: Expected 'mock' to have been called
```

The event is called [here](https://github.com/open-craft/edx-platform/blob/ed8f1e6f5fed34679b7adc9d078dae97243977f9/xmodule/modulestore/mixed.py#L904-L911), but I think the test suit use the base class. Am I right?

Reply:
Hmm.. you're calling `self.store.create_item` here instead of `self.store.create_xblock`, which is where your events are triggered. But I'm not sure where `create_item` is used.. And there's also a `create_child` method too!

I'm running out of time today to look more deeply into this, so could you:

* Add some temporary code on your local devstack that receives the XBLOCK_CREATED signal and uses `logging.error()` or something to log the output to the shell
* Run Studio in your devstack -- will be on port :18010, and you can login as the demo staff user: `staff@example.com` / `edx`
* Create a course, add section, subsection, unit, and a problem block
* Watch the logs through each of these steps to determine where your XBLOCK_CREATED gets triggered.

We want it triggered for sections, subsections, units, and XBocks of all types, so you may have to move the send_event call to the appropriate modulestore method.

Reply:
Thank you, @pomegranited! The method `create_child` was called on all cases: Subsections, Units, and Problem Block.

I changed the event to the `create_child` method and fixed the test to use this.

I will do some more tests, and we are almost done here.